src/jobsautoreport/report.py

```diff
@@ -271,6 +281,30 @@ def _count_cost_by_machine_type(
             ]
         )
 
+    @classmethod
+    def _get_job_type_metrics(
+        cls, usages: list[EquinixUsageEvent], jobs: list[JobDetails]
+    ) -> JobTypeMetrics:
+        return JobTypeMetrics(
+            presubmit=cls._count_cost_by_job_type(usages, jobs, JobType.PRESUBMIT),
+            periodic=cls._count_cost_by_job_type(usages, jobs, JobType.PERIODIC),
+            postsubmit=cls._count_cost_by_job_type(usages, jobs, JobType.POSTSUBMIT),
+            batch=cls._count_cost_by_job_type(usages, jobs, JobType.BATCH),
+        )
+
+    @staticmethod
+    def _count_cost_by_job_type(
+        usages: list[EquinixUsageEvent], jobs: list[JobDetails], job_type: JobType
+    ) -> float:
+        jobs_build_id_to_type = {job.build_id: job.type for job in jobs}
```

Comment:
just get a set where all build_ids are from the selected type
then later in `if usage.job.build_id in jobs_build_id_matching_type`